Added ARG query for A01.40 in network_appdelivery_checklist.en.json

asukumar22 opened the pull request:
Added ARG query for A01.40 in network_appdelivery_checklist.en.json

## Description
Please provide a detailed description of your changes.

## Related Issue
Link to any related issues or discussions here. This helps reviewers understand the context and the need for your changes.

<!-- Please follow this checklist and put an x in each box like this: [x]. -->
## Checklist

- [x] I've tested my changes to ensure they are ready for review.
- [x] I've read the [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) guide.
- [ ] I've updated the documentation (if applicable).
- [x] Resource Graph queries have been included (and tested) for recommendations where ever possible[^note].
- [ ] Resource Graph queries have NOT been included (please explain below).



## Additional Information
Is there any additional context, screenshots, or considerations that might help in the review process? Please include them here.

## Reviewer Notes
Is there a specific area you’d like feedback on? Please highlight it here. We're here to help and learn together! 💡

[^note]:
    Details on how to add Azure Resource Graph queries to recommendations can be found [here](../blob/main/CONTRIBUTING.md#1-adding-or-modifying-resource-graph-queries).

erjosito commented on checklists/network_appdelivery_checklist.en.json:
```diff
@@ -220,7 +220,7 @@
             "ammp": true,
             "severity": "High",
-            "graph": "resources | where type == 'microsoft.network/frontdoorwebapplicationfirewallpolicies' | project policyName=name, policyId=id,policySku=sku.name, links=properties.securityPolicyLinks, enabledState=properties.policySettings.enabledState, mode=properties.policySettings.mode | mvexpand links | extend securityPolicy=links.id | extend securityPolicyParts=split(securityPolicy, '/') ...
+            "graph": "resources | where type == "microsoft.network/applicationgatewaywebapplicationfirewallpolicies" | extend compliant = (properties['policySettings']['mode'] == "Prevention")| where properties['policySettings']['mode'] == "Prevention" | distinct id, name, compliant",
```
* Please use the case-insensitive operator `=~` instead of `==`.
* Please use single quotes for strings inside of the `graph` field, otherwise the JSON is invalid.
Thanks!

asukumar22 commented:
Updated the review comment Jose, sorry =~ was left in type == 'microsoft.network/frontdoorwebapplicationfirewallpolicies', updated it with the =~. Good to be merged now. :)

________________________________
From: Jose Moreno ***@***.***>
Sent: Monday, October 7, 2024 6:36 AM
To: Azure/review-checklists ***@***.***>
Cc: Aarthi Sukumar ***@***.***>; Mention ***@***.***>
Subject: Re: [Azure/review-checklists] Added ARG query for A01.40 in network_appdelivery_checklist.en.json (PR #884)


@asukumar22<https://github.com/asukumar22> could you please resolve the conflicts in this PR? Otherwise it can't be merged

—
Reply to this email directly, view it on GitHub<https://github.com/Azure/review-checklists/pull/884#issuecomment-2396956558>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AQ5NY37VN3F6MROJ653NZW3Z2KE4DAVCNFSM6AAAAABOKRWJXSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGOJWHE2TMNJVHA>.
You are receiving this because you were mentioned.

erjosito commented:
Hey @asukumar22 sorry to bother you again: there seems to be a conflict, possibly with another PR in parallel. It is only a single line one of the "more info" links

asukumar22 commented:
Updated Jose.

Thanks,
Aarthi

________________________________
From: Jose Moreno ***@***.***>
Sent: Monday, October 21, 2024 2:15 PM
To: Azure/review-checklists ***@***.***>
Cc: Aarthi Sukumar ***@***.***>; Mention ***@***.***>
Subject: Re: [Azure/review-checklists] Added ARG query for A01.40 in network_appdelivery_checklist.en.json (PR #884)


Hey @asukumar22<https://github.com/asukumar22> sorry to bother you again: there seems to be a conflict, possibly with another PR in parallel. It is only a single line one of the "more info" links

—
Reply to this email directly, view it on GitHub<https://github.com/Azure/review-checklists/pull/884#issuecomment-2427732560>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AQ5NY377O6OLHEMHJIYDKBDZ4VVGVAVCNFSM6AAAAABOKRWJXSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMRXG4ZTENJWGA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

erjosito commented:
@asukumar22 It looks like there are conflicts to be resolved?
